controllers/cards.js

```diff
@@ -1,47 +1,57 @@
+            .catch(err => res.status(500).json({ message: 'Произошла ошибка в выводе списка карточек', error: err }));
+    },
 
-// создаёт карточку
-const cardPost = (req, res) => {
-    const {name, link} = req.body;
-    Card.create({name, link, owner: req.user._id})
-        .then(card => res.send(card))
-        .catch(err => res.status(500).send({ message: 'Произошла ошибка в создании новой карточки', error: err }));
-};
+    // создаёт карточку
+    cardPost: (req, res) => {
+        const {name, link} = req.body;
+        Card.create({name, link, owner: req.user._id})
+            .then(card => res.send(card))
+            .catch(err => res.status(500).json({ message: 'Произошла ошибка в создании новой карточки', error: err }));
+    },
 
-// удаляет карточку по идентификатору
-const cardDelete = (req, res) => {
-    Card.findByIdAndRemove(req.params.cardId)
-        .then(card => res.send(card))
-        .catch(err => res.status(500).send({ message: 'Произошла ошибка в удалении карточки', error: err }));
-};
+    // удаляет карточку по идентификатору
+    cardDelete: (req, res) => {
+        Card.findOneAndRemove({
+            _id: req.params.id,
+            owner: req.user._id
+        })
+        .then(card => {
+            if (!card) return Promise.reject();
+            res.send(card)
+        })
+        .catch(err => res.status(err ? 404 : 403).json({
+            message: 'Произошла ошибка в удалении карточки',
+            error: err
+        }));
+    },
 
-// поставить лайк карточке
-const cardLikePut = (req, res) => {
-    Card.findByIdAndUpdate(
-        req.params.cardId,
-        { $addToSet: {likes: req.user._id} }, // добавить _id в массив, если его там нет
-        { new: true },
-    )
-    .then(card => res.send(card))
-    .catch(err => res.status(500).send({ message: 'Произошла ошибка в добавлении лайка карточки', error: err }));
-};
+    // поставить лайк карточке
+    cardLikePut: (req, res) => {
+        Card.findByIdAndUpdate(
+            req.params.id,
+            { $addToSet: {likes: req.user._id} }, // добавить _id в массив, если его там нет
+            { new: true }
+        )
+        .orFail()
+        .then(card => res.send(card))
+        .catch(err => res.status(500).json({ message: 'Произошла ошибка в добавлении лайка карточки', error: err }));
+    },
 
-// убрать лайк с карточки
-const cardLikeDelete = (req, res) => {
-    Card.findByIdAndUpdate(
-        req.params.cardId,
-        { $pull: { likes: req.user._id } }, // убрать _id из массива
-        { new: true },
-    )
-    .then(card => res.send(card))
-    .catch(err => res.status(500).send({ message: 'Произошла ошибка в удалении лайка карточки', error: err }));
+    // убрать лайк с карточки
+    cardLikeDelete: (req, res) => {
+        Card.findByIdAndUpdate(
+            req.params.id,
+            { $pull: { likes: req.user._id } }, // убрать _id из массива
+            { new: true }
+        )
+        .then(card => res.send(card))
```

Comment:
в метод `orFail()` можно передать только текст сообщения и потом только ловить в `err.message`, сообщение приходит сюда

> /**
     * Make this query throw an error if no documents match the given `filter`.
     * This is handy for integrating with async/await, because `orFail()` saves you
     * an extra `if` statement to check if no document was found.
     *
     * Example:
     *
     *     // Throws if no doc returned
     *     `await Model.findOne({ foo: 'bar' }).orFail();`
     *
     *     // Throws if no document was updated
     *     `await Model.updateOne({ foo: 'bar' }, { name: 'test' }).orFail();`
     *
     *     // Throws "No docs found!" error if no docs match `{ foo: 'bar' }`
     *     `await Model.find({ foo: 'bar' }).orFail(new Error('No docs found!'));`
     *
     *     // Throws "Not found" error if no document was found
     *     `await Model.findOneAndUpdate({ foo: 'bar' }, { name: 'test' }).
     *       orFail(() => Error('Not found'));`
     *
     * @param err optional error to throw if no docs match `filter`
     */

Reply:
можете на примере показать, не понимаю как вы хотите вызвать функцию в методе `.orFail()` и что потом в ней передать, статус statusCode = 404

Этот статус можно передать только текстом, и что потом, как в `catch` его обработать?

Reply:
Да, конечно. Все очень просто.
Пример:
```js
Model.findOneAndUpdate({ foo: 'bar' }, { name: 'test' })
  .orFail(() => new NotFoundError('No rows found')
  .catch((err) => res.status(err.statusCode || 500).send({ message: err.message }));
```
А как реализовать класс `NotFoundError` я уже писал ранее: https://github.com/karpov2/Mesto-API/pull/1#discussion_r396596481 .